tf-modules/credstash-setup/main.tf

```diff
@@ -41,41 +76,61 @@ resource "aws_dynamodb_table" "credstash-db" {
 }
 
 
-data "aws_caller_identity" "current" {}
+## Writer Policy
 
-data "aws_region" "current" {
-  current = true
+resource "aws_iam_policy" "writer-policy" {
+  count = "${var.create_writer_policy ? 1 : 0}"
+  name = "${var.name_prefix}-credstash-writer"
+  policy = <<END_POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "kms:GenerateDataKey"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_kms_key.credstash-key.arn}"
+    },
+    {
+      "Action": [
+        "dynamodb:PutItem"
+      ],
+      "Effect": "Allow",
+      "Resource": "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/${var.db_table_name}"
+    }
+  ]
 }
-
-
-//KMS Key ARN. It can later be used to store and retrieve secrets:
-// credstash put -k kms_key_arn secret_key secret_value
-// credstash get -k kms_key_arn secret_key
-output "kms_key_arn" {
-  value = "${aws_kms_key.credstash-key.arn}"
+END_POLICY
```

Comment:
I believe this is a place where I'd recommend using the https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html instead of the JSON policy. If it's possible, but not required.